pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper_test.go

```diff
@@ -33,136 +32,157 @@ func TestNewGitLabScraper(t *testing.T) {
+type branchResponse struct {
+	branches     getBranchNamesProjectRepository
+	responseCode int
+}
 
-	case "getBranchNames":
-		if m.err {
-			return errors.New(m.errString)
-		}
-		r := resp.Data.(*getBranchNamesResponse)
-		r.Project.Repository.BranchNames = m.BranchNames
-		r.Project.Repository.RootRef = m.RootRef
+type mrResponse struct {
+	mrs          []getMergeRequestsProjectMergeRequestsMergeRequestConnection
+	page         int
+	responseCode int
+}
 
-	case "getMergeRequests":
-		r := resp.Data.(*getMergeRequestsResponse)
+type contribResponse struct {
+	contribs     []*gitlab.Contributor
+	responseCode int
+}
 
-		if req.Variables.(*__getMergeRequestsInput).State == "opened" {
-			if m.openErr {
-				return errors.New(m.errString)
-			}
-			if len(m.openMergeRequests) == 0 {
-				return nil
-			}
-			r.Project.MergeRequests = m.openMergeRequests[m.curPage]
-			if m.openMergeRequests[m.curPage].PageInfo.HasNextPage == false {
-				m.curPage = 0
-			} else {
-				m.curPage++
+type compareResponse struct {
+	compare      *gitlab.Compare
+	responseCode int
+}
+
+func MockServer(responses *responses) *http.ServeMux {
+	var mux http.ServeMux
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		var reqBody graphql.Request
+		if err := json.NewDecoder(r.Body).Decode(&reqBody); err != nil {
+			return
+		}
+		switch {
+		// These OpNames need to be name of the GraphQL query as defined in genqlient.graphql
+		case reqBody.OpName == "getBranchNames":
+			responses := &responses.branchResponse
+			w.WriteHeader(responses.responseCode)
+			if responses.responseCode == http.StatusOK {
+				branches := getBranchNamesResponse{
+					Project: getBranchNamesProject{
+						Repository: responses.branches,
+					},
+				}
+				graphqlResponse := graphql.Response{Data: &branches}
+				if err := json.NewEncoder(w).Encode(graphqlResponse); err != nil {
+					return
+				}
 			}
+		case reqBody.OpName == "getMergeRequests":
+			responses := &responses.mrResponse
+			w.WriteHeader(responses.responseCode)
+			if responses.responseCode == http.StatusOK {
+				mrs := getMergeRequestsResponse{
+					Project: getMergeRequestsProject{
+						MergeRequests: responses.mrs[responses.page],
+					},
+				}
+				graphqlResponse := graphql.Response{Data: &mrs}
+				if err := json.NewEncoder(w).Encode(graphqlResponse); err != nil {
+					return
+				}
+				responses.page++
 
-		} else if req.Variables.(*__getMergeRequestsInput).State == "merged" {
-			if m.mergedErr {
-				return errors.New(m.errString)
+				//For getCombinedMergeRequests when the first call to getMergeRequests is finished
+				//and the page count needs to be reset for the second or else you get index out of range
+				if len(responses.mrs) == responses.page {
+					responses.page = 0
+				}
 			}
-			if len(m.mergedMergeRequests) == 0 {
-				return nil
+		}
+	})
+	mux.HandleFunc("/api/v4/projects/project/repository/contributors", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Do we want to "abstract" out the endpoints to constants instead of inserting them here? I feel as though we'd want something like this instead:

```go
type Endpoints struct {
    ProjectRepositoryContributors string
}

var (
	endpoints := Endpoints{
        ProjectRepositoryContributors: "/api/v4/projects/project/repository/contributors",
	}
)
//...
mux.HandleFunc(endpoints.ProjectRepositoryContributors, func(w http.ResponseWriter, r *http.Request) {
```

Reply:
Will update this in a subsequent pr, for now they don't need to change